lms/djangoapps/course_api/blocks/api.py

```diff
@@ -43,24 +45,21 @@ def get_blocks(
         return_type (string): Possible values are 'dict' or 'list'. Indicates
             the format for returning the blocks.
     """
-    # construct BlocksAPITransformer
-    blocks_api_transformer = BlocksAPITransformer(
-        block_counts,
-        student_view_data,
-        depth,
-        nav_depth
-    )
-
-    # list of transformers to apply, adding user-specific ones if user is provided
-    transformers = [blocks_api_transformer]
+    # create ordered list of transformers, adding BlocksAPITransformer at end.
+    transformers = BlockStructureTransformers()
     if user is not None:
         transformers += COURSE_BLOCK_ACCESS_TRANSFORMERS + [ProctoredExamTransformer()]
+    transformers += [
+        BlocksAPITransformer(
+            block_counts,
+            student_view_data,
+            depth,
+            nav_depth
+        )
+    ]
```

Comment:
BTW, why have different lists of transformers in this situation? Since transformers do different things based on the user anyway, couldn't we just always have one list of transformers (and leave it to the transforms to perform no-ops if they want)?